astro/src/content/blog/custom-scopes-in-third-party-applications.mdx

```diff
@@ -0,0 +1,96 @@
+
+<RemoteCode url="https://raw.githubusercontent.com/FusionAuth/fusionauth-example-ruby-on-rails-custom-scopes/main/fusionauth-quickstart-ruby-on-rails-api-modifications/app/controllers/get_balance_controller.rb" lang="ruby" />
+
+## Budget Buddy Integration
+
+To make this example simple and demonstrate that the Budget Buddy application uses a completely separate identity provider, the login has been set up to accept any email.
+
+![Budget Buddy Log In.](/img/blogs/custom-scopes-in-third-party-applications/budget-buddy-login.png)
+
+Once logged in, the user can choose to see their balance. Notice they will first need to connect to Change Bank to allow this.
+
+![Budget Buddy Connect to Change Bank.](/img/blogs/custom-scopes-in-third-party-applications/budget-buddy-get-balance-start.png)
+
+When the user clicks connect, they will be prompted for a Change Bank login. Here they must enter their username and password for the Change Bank application.
+
+<Aside type="note">
+Behind the scenes, Budget Buddy uses [OmniAuth](https://rubygems.org/gems/omniauth) to help with the OAuth login flow for Change Bank.  Again, this is beyond the scope of this blog. For further details please see the [FusionAuth Custom Scopes Example Application](https://github.com/FusionAuth/fusionauth-example-ruby-on-rails-custom-scopes)
+</Aside>
+
+![Change Bank Login.](/img/blogs/custom-scopes-in-third-party-applications/change-bank-login.png)
+
+After the user logs in, they will be presented with consent options. This allows the user to choose what Change Bank information and actions the Budget Buddy application has access to. In this case, the `Allow Budget Buddy to read your Change Bank balance` consent grants the `balance:read` scope.
+
+![Change Bank Consent.](/img/blogs/custom-scopes-in-third-party-applications/change-bank-consent.png)
+
+When the user allows the consents, Budget Buddy will be able to read the user's Change Bank balance.
+
+![Budget Buddy Read Change Bank Balance.](/img/blogs/custom-scopes-in-third-party-applications/budget-buddy-balance-read.png)
+
+From here the user can choose to disconnect Change Bank from the Budget Buddy application or leave it connected. As long as Budget Buddy has a valid authorization token it will be allowed to read the user's balance from the Change Bank APIs.
+
+## Conclusion
+
+It can be a little confusing with the screen shots all showing the local host due to the way Safari renders the URL. In the example application, there are actually two applications (Change Bank API / Budget Buddy Web App) and FusionAuth is the IdP for the Change Bank applications. In a real world example, Change Bank would likely have a web application as well. A diagram of what that may look like would be as follows: 
```

Comment:
```suggestion
It can be confusing with the screenshots all showing the local host due to the way Safari renders the URL. In the example application, there are two applications (Change Bank API / Budget Buddy Web App) and FusionAuth is the IdP for the Change Bank applications. In a real-world example, Change Bank would likely also have a web application. A diagram of what that may look like would be as follows: 
```